biz.aQute.bndlib/src/aQute/bnd/buildtool/ToolManager.java

```diff
@@ -0,0 +1,172 @@
+package aQute.bnd.buildtool;
+
+import java.io.File;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.attribute.PosixFilePermission;
+import java.time.Instant;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.BiConsumer;
+
+import aQute.bnd.build.Workspace;
+import aQute.bnd.header.Attrs;
+import aQute.bnd.http.HttpClient;
+import aQute.bnd.osgi.EmbeddedResource;
+import aQute.bnd.osgi.Instruction;
+import aQute.bnd.osgi.Instructions;
+import aQute.bnd.osgi.Jar;
+import aQute.bnd.osgi.Processor;
+import aQute.bnd.osgi.Resource;
+import aQute.bnd.osgi.Verifier;
+import aQute.lib.io.IO;
+import aQute.lib.strings.Strings;
+import aQute.lib.utf8properties.UTF8Properties;
+
+/**
+ * Install gradle from a repository in this workspace.
+ */
+public class ToolManager extends Processor {
+
+	final Workspace						workspace;
+	final BiConsumer<String, Object[]>	reporter;
+
+	public ToolManager(Workspace workspace, BiConsumer<String, Object[]> reporter) {
+		super(workspace);
+		this.reporter = reporter;
+		use(workspace);
+		this.workspace = workspace;
+		setBase(workspace.getBase());
+	}
+
+	public String install(String url, Map<String, String> parameters, boolean force) {
+		HttpClient client = workspace.getPlugin(HttpClient.class);
+		try {
+			if (client.isOffline()) {
+				return "The workspace is off line";
+			}
+
+			if (Verifier.isVersion(url)) {
+				aQute.bnd.version.Version v = new aQute.bnd.version.Version(url);
+				url = "https://github.com/bndtools/gradle.workspace/archive/refs/tags/" + v.toStringWithoutQualifier()
```

Comment:
This is another new repo? It does not match the pattern above `bndtools/workspace.tool.?`.

Reply:
mistake, left over, dead code